python/nano/tutorial/notebook/training/pytorch/convert_pytorch_training_torchnano.ipynb

```diff
@@ -0,0 +1,285 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "nbsphinx": "hidden"
+   },
+   "outputs": [],
+   "source": [
+    "!pip install --pre --upgrade bigdl-nano[pytorch] # install the nightly-bulit version\n",
+    "!source bigdl-nano-init # set environment variables"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "> 📝 **Note**\n",
+    ">\n",
+    "> Before starting your PyTorch application, it is highly recommended to run `source bigdl-nano-init` to set several environment variables based on your current hardware. Empirically, these variables will bring big performance increase for most PyTorch applications on training workloads."
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## PyTorch Training Loops Example"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Suppose you would like to finetune a [ResNet-18 model](https://pytorch.org/vision/main/models/generated/torchvision.models.resnet18.html) (pretrained on ImageNet dataset) on [OxfordIIITPet dataset](https://pytorch.org/vision/main/generated/torchvision.datasets.OxfordIIITPet.html), you may create datasets, the model and define your training loops as follows:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "nbsphinx": "hidden"
+   },
+   "outputs": [],
+   "source": [
+    "# Define model and dataloaders\n",
+    "\n",
+    "from torch import nn\n",
+    "from torchvision.models import resnet18\n",
+    "\n",
+    "class MyPytorchModule(nn.Module):\n",
+    "    def __init__(self):\n",
+    "        super().__init__()\n",
+    "        self.model = resnet18(pretrained=True)\n",
+    "        num_ftrs = self.model.fc.in_features\n",
+    "        # Here the size of each output sample is set to 37.\n",
+    "        self.model.fc = nn.Linear(num_ftrs, 37)\n",
+    "\n",
+    "    def forward(self, x):\n",
+    "        return self.model(x)\n",
+    "\n",
+    "\n",
+    "import torch\n",
+    "from torchvision import transforms\n",
+    "from torchvision.datasets import OxfordIIITPet\n",
+    "from torch.utils.data.dataloader import DataLoader\n",
+    "\n",
+    "def create_dataloaders():\n",
+    "    train_transform = transforms.Compose([transforms.Resize(256),\n",
+    "                                          transforms.RandomCrop(224),\n",
+    "                                          transforms.RandomHorizontalFlip(),\n",
+    "                                          transforms.ColorJitter(brightness=.5, hue=.3),\n",
+    "                                          transforms.ToTensor(),\n",
+    "                                          transforms.Normalize([0.485, 0.456, 0.406],\n",
+    "                                                               [0.229, 0.224, 0.225])])\n",
+    "    val_transform = transforms.Compose([transforms.Resize(256),\n",
```

Comment:
We can remove the code creating `val_dataset` and `val_dataloader` because we don't use them

Reply:
Although `val_dataloder` are not used directly, I feel like it could be useful to explain the usage of `setup` function?

Reply:
We may change the input format or output format when there are more than one optimziers or dataloaders, so I think we can only show single optimizer and single dataloader here